demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js

```diff
@@ -146,17 +147,19 @@ document.addEventListener("DOMContentLoaded", function () {
             
             updateRewardValue(reward, articleDoc);
 
-            gaugeInterval = setInterval(function () {
-                reward -= RewardDecreaseAmount;
-                if (reward <= RewardDecreaseLimit) {
-                    clearInterval(gaugeInterval);
-                    gaugeInterval = -1;
-                    updateRewardValue(RewardDecreaseLimit, articleDoc);
-                } else {
-                    updateRewardValue(reward, articleDoc);
-                }
-
-            }, RewardDecreaseInterval*1000);
+            gauge.addEventListener("transitionend", function (event) {
+                gaugeInterval = setInterval(function () {

```

Comment:
Remove the event listener as soon as the handler is called to avoid any possible issues with multiple firing